CommandAPI/commandapi-core/src/main/java/dev/jorel/commandapi/CommandAPICommand.java

```diff
@@ -295,7 +303,81 @@ public CommandAPICommand executesNative(NativeResultingCommandExecutor executor)
 		this.executor.addResultingExecutor(executor);
 		return this;
 	}
-	
+
+	//================================================================================
+	//endregion
+	//region Set (Sets the underlying array/etc. DIRECTLY) Commands
+	//Note: We should probably improve the documentation in this area too, to better reflect what it does.
+	// (Many people may just think its the plural version of the `with` commands
+	//================================================================================
+
+
+	/**
+	 * Sets the permissions required to run this command
+	 * @param permission the permission required to run this command
+	 * @return this command builder
+	 * @deprecated setPermission and withPermission function exactly the same
+	 */
+	public CommandAPICommand setPermission(CommandPermission permission) {
```

Comment:
`setPermission` is not deprecated. Even though its functionally equivalent to `setPermission`, it has different semantics. `withPermission` is designed to be used by the builder, whereas `setPermission` is a setter, with its corresponding getter method `getPermission`